autoload/lsc/server.vim

```diff
@@ -87,6 +87,7 @@ function! s:Kill(server, status, OnExit) abort
       call a:server._channel.notify('exit', v:null)
     endif
     if a:OnExit != v:null | call a:OnExit() | endif
+    call lsc#config#unmapKeys()
```

Comment:
Looks like this will unmap keys for whichever buffer is current when the server exists - there isn't a guarantee that the current buffer is one handled by the server which exited, and we'll missed unmapping keys on background buffers as well.

I think we'll need to loop over buffers for each filetype and unmap in each of them which makes things a little trickier.

Reply:
You're right, this is tricky because as far as I know Vim doesn't provide a method for (un)mapping buffer-local keys for buffers other than the current one. Which means to unmap the buffer-local keys in all buffers associated with the language server, we would either need to:

1. Iterate through each buffer and unmap the keys (probably slow and requires switching buffers which has all other kinds of pitfalls)

2. Hook into the BufEnter autocmd and check the status of the server associated with that buffer's filetype: if the status is disabled then unmap the keys. This essentially unmaps "on demand".

Unless there is something I'm overlooking then it seems to me that option #2 is the clear winner. I will update the PR with these changes for your review.